app/connectors/AddressLookupConnector.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.FrontendAppConfig
+import javax.inject.Inject
+import models.AddressLookup
+import play.api.Logger
+import play.api.http.Status._
+import play.api.libs.json.Reads
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AddressLookupConnector @Inject()(http: HttpClient, config: FrontendAppConfig) {
+
+  def addressLookupByPostcode(postCode: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[AddressLookup]] = {
+
+    val addressLookupUrl: String = s"${config.addressLookUpUrl}/v2/uk/addresses"
+    val urlParams = Seq ("postcode" -> postCode)
+
+    implicit val reads: Reads[Seq[AddressLookup]] = AddressLookup.addressesLookupReads
+
+    http.GET[HttpResponse](
+      url = addressLookupUrl,
+      queryParams = urlParams,
+      headers = Seq("X-Hmrc-Origin" -> "DAC6")
+    )(implicitly, implicitly, implicitly) flatMap {
```

Comment:
if you're passing everything with the implicitly method it means the compiler should be able to resolve them all too